iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/evaluator/CompactionEvaluator.java

```diff
@@ -89,11 +90,27 @@ public boolean isEligibleForCompaction() {
         return isMinorNecessary();
       case MAJOR:
         return isMajorNecessary();
+      case SMART:
+        return isMinorNecessary() || isMajorNecessary();
       default:
         return false;
     }
   }
 
+  public CompactionType determineCompactionType() {
+    if (ci.type == CompactionType.SMART) {
+      if (isMajorNecessary()) {
+        return CompactionType.MAJOR;
+      } else if (isMinorNecessary()) {
+        return CompactionType.MINOR;
+      } else {
+        throw new RuntimeException("Unknown compaction type");
```

Comment:
Return a null instead of throwing an exception.

Reply:
why?